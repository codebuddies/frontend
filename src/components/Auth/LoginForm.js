import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Link, Redirect } from 'react-router-dom';
import { Box, Button, TextField } from '@material-ui/core/';
import { useAuth } from './AuthContext';
import axios from 'axios';

const LoginForm = ({ toggleActiveForm }) => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const [errorMessage, setErrorMessage] = useState('');
  const auth = useAuth();
  const referer = '/profile';

  const handleLogin = e => {
    e.preventDefault();
    const data = {
      username,
      password,
    };
    axios
      .post('/auth/obtain_token/', data)
      .then(res => {
        auth.setAuthTokens(res.data);
        setIsLoggedIn(true);
      })
      .catch(error => {
        if (error.response) {
          setErrorMessage(error.response.data);
        } else {
          setErrorMessage('There was an error!');
        }
      });
  };

  if (isLoggedIn) {
    return <Redirect to={referer} />;
  }
  return (
    <>
      {(auth && auth.authTokens) || isLoggedIn ? (
        <p>Welcome!</p>
      ) : (
        <Box
          component="form"
          display="flex"
          flexWrap="wrap"
          noValidate
          autoComplete="off"
          onSubmit={handleLogin}
          data-testid="loginForm"
        >
          <Box component="h1" fontSize={18}>
            Log in
          </Box>
          <TextField
            id="username"
            label="username"
            fullWidth
            required
            variant="outlined"
            margin="dense"
            type="text"
            onChange={e => setUsername(e.target.value)}
          />
          <TextField
            id="password"
            label="Password"
            fullWidth
            required
            variant="outlined"
            margin="dense"
            type="password"
            onChange={e => setPassword(e.target.value)}
          />

          <Box color="error.main">{errorMessage && errorMessage}</Box>

          <Box width="100%" marginTop={2}>
            <Button variant="contained" color="primary" type="submit">
              Log in
            </Button>
          </Box>

          <p>
            Need an account?{' '}
            {toggleActiveForm ? (
              <Box
                component="button"
                color="primary.main"
                padding={0}
                marginLeft={1}
                border={0}
                bgcolor="transparent"
                fontSize={16}
                onClick={toggleActiveForm}
              >
                Sign up
              </Box>
            ) : (
              <Link to="/signup">Sign up</Link>
            )}
            .
          </p>
        </Box>
      )}
    </>
  );
};

const { func } = PropTypes;

LoginForm.propTypes = {
  toggleActiveForm: func,
};

export default LoginForm;
