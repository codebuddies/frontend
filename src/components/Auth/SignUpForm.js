import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Link, Redirect } from 'react-router-dom';
import { Box, Button, TextField } from '@material-ui/core/';
import axios from 'axios';
import { useAuth } from './AuthContext';

const SignUpForm = ({ toggleActiveForm }) => {
  const [firstName, setFirstName] = useState(null);
  const [lastName, setLastName] = useState(null);
  const [username, setUsername] = useState(null);
  const [email, setEmail] = useState(null);
  const [password, setPassword] = useState(null);
  const [errorMessage, setErrorMessage] = useState(null);
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const referer = '/profile';
  const auth = useAuth();

  const handleSignup = e => {
    e.preventDefault();
    const data = {
      username: username,
      password: password,
      first_name: firstName,
      last_name: lastName,
      email: email,
    };
    axios
      .post('/auth/users/', data)
      .then(res => {
        auth.setAuthTokens(res.data);
        setIsLoggedIn(true);
        setErrorMessage('');
      })
      .catch(err => {
        setErrorMessage(Object.values(err.response.data).join(''));
      });
  };
  if (isLoggedIn) {
    return <Redirect to={referer} />;
  }
  return (
    <>
      {(auth && auth.authTokens) || isLoggedIn ? (
        <p>Welcome!</p>
      ) : (
        <>
          <Box
            component="form"
            display="flex"
            flexWrap="wrap"
            noValidate
            autoComplete="off"
            onSubmit={handleSignup}
            data-testid="signupForm"
          >
            <Box component="h1" fontSize={18}>
              Create an account
            </Box>
            <TextField
              id="first-name"
              label="First Name"
              fullWidth
              variant="outlined"
              margin="dense"
              onChange={e => setFirstName(e.target.value)}
            />
            <TextField
              id="last-name"
              label="Last Name"
              fullWidth
              variant="outlined"
              margin="dense"
              onChange={e => setLastName(e.target.value)}
            />
            <TextField
              id="username"
              label="Username"
              fullWidth
              required
              variant="outlined"
              margin="dense"
              onChange={e => setUsername(e.target.value)}
            />
            <TextField
              id="email"
              label="Email"
              fullWidth
              required
              variant="outlined"
              margin="dense"
              type="email"
              onChange={e => setEmail(e.target.value)}
            />
            <TextField
              id="password"
              label="Password"
              fullWidth
              required
              variant="outlined"
              margin="dense"
              type="password"
              onChange={e => setPassword(e.target.value)}
            />
            {errorMessage && <Box color="error.main"> {errorMessage}</Box>}
            <Box width="100%" marginTop={2}>
              <Button variant="contained" color="primary" type="submit">
                Sign Up
              </Button>
            </Box>
            <p>
              Already have an account?
              {toggleActiveForm ? (
                <Box
                  component="button"
                  color="primary.main"
                  padding={0}
                  marginLeft={1}
                  border={0}
                  bgcolor="transparent"
                  fontSize={16}
                  onClick={toggleActiveForm}
                >
                  Log in
                </Box>
              ) : (
                <Link to="/login"> Log in</Link>
              )}
              .
            </p>
          </Box>
        </>
      )}
    </>
  );
};

const { func } = PropTypes;

SignUpForm.propTypes = {
  toggleActiveForm: func,
};

export default SignUpForm;
