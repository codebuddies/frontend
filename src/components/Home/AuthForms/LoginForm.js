import React from 'react';
import PropTypes from 'prop-types';
import { Box, Button, TextField } from '@material-ui/core/';

const LoginForm = ({ toggleActiveForm }) => {
  return (
    <Box
      component="form"
      display="flex"
      flexWrap="wrap"
      noValidate
      autoComplete="off"
    >
      <Box component="h1" fontSize={18}>
        Log in
      </Box>
      <TextField
        id="email"
        label="Email"
        fullWidth
        required
        variant="outlined"
        margin="dense"
        type="email"
      />
      <TextField
        id="password"
        label="Password"
        fullWidth
        required
        variant="outlined"
        margin="dense"
        type="password"
      />

      <Box width="100%" marginTop={2}>
        <Button variant="contained" color="primary">
          Log in
        </Button>
      </Box>

      <p>
        Need an account?{' '}
        <Box
          component="button"
          color="primary.main"
          padding={0}
          marginLeft={1}
          border={0}
          bgcolor="transparent"
          fontSize={16}
          onClick={toggleActiveForm}
        >
          Sign up
        </Box>
        .
      </p>
    </Box>
  );
};

const { func } = PropTypes;

LoginForm.propTypes = {
  toggleActiveForm: func.isRequired,
};

export default LoginForm;
