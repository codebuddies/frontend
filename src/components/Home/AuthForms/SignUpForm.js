import React from 'react';
import PropTypes from 'prop-types';
import { Box, Button, TextField } from '@material-ui/core/';

const SignUpForm = ({ toggleActiveForm }) => {
  return (
    <Box
      component="form"
      display="flex"
      flexWrap="wrap"
      noValidate
      autoComplete="off"
    >
      <Box component="h1" fontSize={18}>
        Create an account
      </Box>
      <TextField
        id="first-name"
        label="First Name"
        fullWidth
        variant="outlined"
        margin="dense"
      />
      <TextField
        id="last-name"
        label="Last Name"
        fullWidth
        variant="outlined"
        margin="dense"
      />
      <TextField
        id="username"
        label="Username"
        fullWidth
        required
        variant="outlined"
        margin="dense"
      />
      <TextField
        id="email"
        label="Email"
        fullWidth
        required
        variant="outlined"
        margin="dense"
        type="email"
      />
      <TextField
        id="password"
        label="Password"
        fullWidth
        required
        variant="outlined"
        margin="dense"
        type="password"
      />

      <Box width="100%" marginTop={2}>
        <Button variant="contained" color="primary">
          Sign Up
        </Button>
      </Box>

      <p>
        Already have an account?
        <Box
          component="button"
          color="primary.main"
          padding={0}
          marginLeft={1}
          border={0}
          bgcolor="transparent"
          fontSize={16}
          onClick={toggleActiveForm}
        >
          Log in
        </Box>
        .
      </p>
    </Box>
  );
};

const { func } = PropTypes;

SignUpForm.propTypes = {
  toggleActiveForm: func.isRequired,
};

export default SignUpForm;
