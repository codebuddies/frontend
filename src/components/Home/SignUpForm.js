import React from 'react';
import { Box, Button, TextField } from '@material-ui/core/';

const SignUpForm = () => {
  return (
    <Box
      component="form"
      display="flex"
      flexWrap="wrap"
      noValidate
      autoComplete="off"
    >
      <TextField
        id="first-name"
        label="First Name"
        fullWidth
        variant="outlined"
        margin="dense"
      />
      <TextField
        id="last-name"
        label="Last Name"
        fullWidth
        variant="outlined"
        margin="dense"
      />
      <TextField
        id="username"
        label="Username"
        fullWidth
        required
        variant="outlined"
        margin="dense"
      />
      <TextField
        id="email"
        label="Email"
        fullWidth
        required
        variant="outlined"
        margin="dense"
        type="email"
      />
      <TextField
        id="password"
        label="Password"
        fullWidth
        required
        variant="outlined"
        margin="dense"
        type="password"
      />

      <Button variant="contained" color="primary">
        Sign Up
      </Button>

      <p>Already have an account? Log in.</p>
    </Box>
  );
};

export default SignUpForm;
