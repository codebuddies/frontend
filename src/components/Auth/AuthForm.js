import React, { useState } from 'react';
import { Box, Paper } from '@material-ui/core/';
import SignUpForm from './components/SignUpForm';
import LoginForm from './components/LoginForm';

const AuthForm = () => {
  const [activeSignUpForm, setActiveSignUpform] = useState(true);

  const toggleActiveForm = event => {
    event.preventDefault();
    setActiveSignUpform(!activeSignUpForm);
  };

  return (
    <Box component={Paper} elevation={3} padding={3}>
      {activeSignUpForm ? (
        <SignUpForm toggleActiveForm={toggleActiveForm} />
      ) : (
        <LoginForm toggleActiveForm={toggleActiveForm} />
      )}
    </Box>
  );
};

export default AuthForm;
