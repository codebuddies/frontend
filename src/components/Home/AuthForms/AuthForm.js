import React, { useState } from 'react';
import { Box, Paper } from '@material-ui/core/';
import SignUpForm from './SignUpForm';
import LoginForm from './LoginForm';

const AuthForm = () => {
  const [activeSignUpForm, setActiveSignUpform] = useState(true);

  const toggleActiveForm = event => {
    event.preventDefault();
    setActiveSignUpform(!activeSignUpForm);
  };

  return (
    <Box component={Paper} elevetion={3} padding={3}>
      {activeSignUpForm && <SignUpForm toggleActiveForm={toggleActiveForm} />}
      {!activeSignUpForm && <LoginForm toggleActiveForm={toggleActiveForm} />}
    </Box>
  );
};

export default AuthForm;
