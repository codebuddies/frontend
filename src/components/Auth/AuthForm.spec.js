import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter } from 'react-router-dom';
import AuthForm from './AuthForm';
import SignUpForm from './SignUpForm';
import LoginForm from './LoginForm';

describe('AuthForm', () => {
  it('should render accordingly', () => {
    const { asFragment } = render(<AuthForm />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should only show the Sign Up form by default', () => {
    const { getByText, queryByTestId } = render(<AuthForm />);

    expect(getByText('Create an account')).toBeInTheDocument();
    expect(queryByTestId('loginForm')).toBeNull();
  });

  describe('when clicking on the Log in button in the Sign Up Form', () => {
    it('should show the Login Form', () => {
      const { getByText, queryByTestId, getByTestId } = render(<AuthForm />);
      fireEvent.click(getByText('Log in'));
      expect(getByTestId('loginForm')).toBeInTheDocument();
      expect(queryByTestId('signupForm')).toBeNull();
    });
  });

  describe('when clicking on the Sign up button in the Log In Form', () => {
    it('should show the Sign Up form', () => {
      const { getByText, queryByTestId } = render(<AuthForm />);
      fireEvent.click(getByText('Log in'));
      fireEvent.click(getByText('Sign up'));
      expect(queryByTestId('loginForm')).toBeNull();
      expect(getByText('Create an account')).toBeInTheDocument();
    });
  });
});

describe('Signup', () => {
  it('Fill out the signup form', () => {
    const { getByText, debug, getByRole } = render(
      <BrowserRouter>
        <SignUpForm />
      </BrowserRouter>
    );
    const mockRegisterCall = jest.fn();
    mockRegisterCall.mockResolvedValueOnce({
      data: {
        username: 'Carolyne.Carter',
        token:
          'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6IkNhcm9seW5lLkNhcnRlciIsImlhdCI6MTU4NDMzODQ4NiwiZXhwIjoxNTg0MzQyMDg2LCJ1c2VyX2lkIjo4MCwib3JpZ19pYXQiOjE1ODQzMzg0ODZ9.saO6OCOKV1uwHjTbM-iDGmhbkMNCnzrGFj4TBYnTv2E',
        first_name: 'Bud',
        last_name: 'Paucek',
        email: 'Josiah.McLaughlin8@yahoo.com',
      },
    });
    debug();
    const submit = getByRole('button');
    fireEvent.click(submit);
    expect(getByText('Create an account')).toBeInTheDocument();
  });
});

describe('Login', () => {
  it('Fill out the login form', () => {
    const { getAllByText, debug, getByRole } = render(
      <BrowserRouter>
        <LoginForm />
      </BrowserRouter>
    );
    const mockLoginCall = jest.fn();
    mockLoginCall.mockResolvedValueOnce({
      data: {
        token:
          'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VybmFtZSI6IkNhcm9seW5lLkNhcnRlciIsImlhdCI6MTU4NDM0MDAyMiwiZXhwIjoxNTg0MzQzNjIyLCJ1c2VyX2lkIjo4MCwib3JpZ19pYXQiOjE1ODQzNDAwMjJ9.0zNlXPVAjkBjxUQjq4B0HXnvrez93H2pz6n2ROKWzzg',
        username: 'Carolyne.Carter',
      },
    });
    debug();
    const submit = getByRole('button');
    fireEvent.click(submit);
    expect(getAllByText('Log in')[1]).toBeInTheDocument();
  });
});
