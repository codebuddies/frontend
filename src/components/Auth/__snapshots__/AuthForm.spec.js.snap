// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm should render accordingly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation3 MuiBox-root MuiBox-root-2 MuiPaper-rounded"
  >
    <form
      autocomplete="off"
      class="MuiBox-root MuiBox-root-30"
      data-testid="signupForm"
      novalidate=""
    >
      <h1
        class="MuiBox-root MuiBox-root-31"
      >
        Create an account
      </h1>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
          data-shrink="false"
          for="first-name"
        >
          First Name
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            id="first-name"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-91 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-92"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
          data-shrink="false"
          for="last-name"
        >
          Last Name
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            id="last-name"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-91 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-92"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="false"
          for="username"
        >
          Username
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            id="username"
            required=""
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-91 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-92"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="false"
          for="email"
        >
          Email
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            id="email"
            required=""
            type="email"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-91 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-92"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root MuiFormControl-marginDense MuiFormControl-fullWidth"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined Mui-required Mui-required"
          data-shrink="false"
          for="password"
        >
          Password
          <span
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            id="password"
            required=""
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-91 MuiOutlinedInput-notchedOutline"
            style="padding-left: 8px;"
          >
            <legend
              class="PrivateNotchedOutline-legend-92"
              style="width: 0.01px;"
            >
              <span>
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-93"
      />
      <div
        class="MuiBox-root MuiBox-root-94"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Sign Up
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <p>
        Already have an account?
        <button
          class="MuiBox-root MuiBox-root-126"
        >
          Log in
        </button>
        .
      </p>
    </form>
  </div>
</DocumentFragment>
`;
